src/mono/wasm/runtime/pthreads/worker/index.ts

```diff
@@ -23,40 +25,57 @@ export {
     WorkerThreadEventTarget,
 } from "./events";
 
+class WorkerSelf implements PThreadSelf {
+    readonly isBrowserThread = false;
+    constructor(readonly pthread_id: pthread_ptr, readonly portToBrowser: MessagePort) { }
+    postMessageToBrowser(message: MonoThreadMessage, transfer?: Transferable[]) {
+        if (transfer) {
+            this.portToBrowser.postMessage(message, transfer);
+        } else {
+            this.portToBrowser.postMessage(message);
+        }
+    }
+    addEventListenerFromBrowser(listener: (event: MessageEvent<MonoThreadMessage>) => void) {
+        this.portToBrowser.addEventListener("message", listener);
+    }
+}
+
+// we are lying that this is never null, but afterThreadInit should be the first time we get to run any code
+// in the worker, so this becomes non-null very early.
+export let pthread_self: PThreadSelf = null as any as PThreadSelf;
+
 /// This is the "public internal" API for runtime subsystems that wish to be notified about
 /// pthreads that are running on the current worker.
 /// Example:
 ///    currentWorkerThreadEvents.addEventListener(dotnetPthreadCreated, (ev: WorkerThreadEvent) => {
-///       console.debug ("thread created on worker with id", ev.pthread_ptr);
+///       console.debug("MONO_WASM: thread created on worker with id", ev.pthread_ptr);
 ///    });
 export const currentWorkerThreadEvents: WorkerThreadEventTarget =
     MonoWasmThreads ? new EventTarget() : null as any as WorkerThreadEventTarget; // treeshake if threads are disabled
 
 function monoDedicatedChannelMessageFromMainToWorker(event: MessageEvent<string>): void {
-    console.debug("got message from main on the dedicated channel", event.data);
+    console.debug("MONO_WASM: got message from main on the dedicated channel", event.data);
```

Comment:
After we merge both this and my PR, we could prefix this with `if (runtimeHelpers.diagnostic_tracing)` so that by default the runtime is not noisy. It matters in console like nodeJS. @maraf suggested it's about a time when we introduce logging component to centralize all this logic to one place.

Reply:
agreed